src/types.ts

```diff
@@ -1,16 +1,20 @@
 import { AbiItem } from 'viem';
+import { SortClosure, WhereClosure } from './database/filters';
 
 export type Hash = `0x${string}`;
 
-export interface PersistanceObject {
+export interface PersistenceObject {
   init(): Promise<void>;
   saveBatch(batch: EventLog[], blockNumber?: bigint): Promise<void>;
   getLatestIndexedBlockForChain(chainId: number): Promise<number | undefined>;
-  getJsonObjectPropertySqlFragment(column: string, propertyName: string): string;
-  queryAll<T>(query: string): Promise<T[]>;
-  queryOne<T>(query: string): Promise<T>;
-  queryRun(query: string): Promise<void>;
   disconnect(): Promise<void>;
+  filter<T extends {}>(
+    table: string,
+    whereClosures: WhereClosure[],
+    sortClosures: SortClosure[],
+    limit: number,
+    offset: number,
+  ): Promise<T[]> ;
```

Comment:
I don't like that this function accepts many parameters, let's swap that with an object and set some reasonable defaults for every param e.g.:
```ts
 filter<T>({
    table,
    whereClauses = [],
    sortClauses = [],
    limit = 100,
    offset = 0,
  }: {
    table: string;
    whereClauses: WhereClause[];
    sortClauses: SortClause[];
    limit: number;
    offset: number;
  }): Promise<T[]>;
```